VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitVideoFragment.java

```diff
@@ -216,6 +218,7 @@ public void setUserVisibleHint(boolean isVisibleToUser) {
         if ( ViewPagerDownloadManager.instance.inInitialPhase(unit) )
             return;
         if (isVisibleToUser) {
+            checkVideoStatus(unit);
```

Comment:
Its in the right place imo, cuz before showing the wifi message we need to see if the video is downloaded or not and act accordingly (which is what this function does).